app/src/main/java/org/p2p/wallet/svl/interactor/ReceiveViaLinkInteractor.kt

```diff
@@ -0,0 +1,38 @@
+package org.p2p.wallet.svl.interactor
+
+import java.math.BigInteger
+import org.p2p.core.utils.isMoreThan
+import org.p2p.wallet.newsend.model.TemporaryAccount
+import org.p2p.wallet.rpc.repository.account.RpcAccountRepository
+import org.p2p.wallet.svl.model.SendLinkGenerator
+import org.p2p.wallet.svl.model.TemporaryAccountState
+import org.p2p.wallet.user.repository.UserLocalRepository
+
+class ReceiveViaLinkInteractor(
+    private val rpcAccountRepository: RpcAccountRepository,
+    private val userLocalRepository: UserLocalRepository
+) {
+
+    suspend fun parseLink(link: SendViaLinkWrapper): TemporaryAccountState {
+        val temporaryAccount = SendLinkGenerator.parseTemporaryAccount(link)
```

Comment:
SendLinkGenerator.parseAccountfromLink(link) ?
because parseTemporaryAccount means that we pass temporary account and parse it